### PR TITLE
Am 2762 sscs wa mapping rules staff

### PR DESCRIPTION
**Pull request by abul-syed**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-2762

### Change description ###

SSCS_WA - Add mapping rules in ORM for STAFF users

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by adityadwadasi — APPROVED** [review]

**Review by tomxelliott — APPROVED** [review]
Code, looks good. There seems to be a CVE issue blocking the build. I have updated with base branch so hopefully that will resolve the issue by pulling in latest changes

**Review by mattnayler — APPROVED** [review]
Additional review comments relate t ounit-tests only so can be actioned under patch to `SSCS_WA_1_0` feture branch.

File: src/main/resources/validationrules/core/hearing-role-admin-global.drl [hunk 1]
```diff
@@ -26,7 +26,7 @@ import function uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMapping
 rule "global_hearing_manager_admin"
 when
   $f:  FeatureFlag(status && flagName == FeatureFlagEnum.SSCS_HEARING_1_0.getValue())
-  $up: CaseWorkerAccessProfile((roleId in ("3","4","5","9","10","12","13") && serviceCode in ("BBA3"))
+  $up: CaseWorkerAccessProfile((roleId in ("3","4","5","12","13") && serviceCode in ("BBA3"))
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/main/resources/validationrules/core/hearing-role-admin-global.drl:29]: tests updated, in #1493

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java [hunk 2]
```diff
@@ -252,7 +259,8 @@ void shouldReturnSscsAdminOrgRolesForRoleId() {
                         roleAssignment.getAttributes().get("workTypes").asText());
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java:259]: have removed as cases has been covered in DroolSscsStaffOrgRolesTest, see #1493

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java [hunk 3]
```diff
@@ -294,7 +309,8 @@ void shouldReturnSscsAdminOrgRolesForRoleId_caseAndTaskSupervisor() {
...
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java:309]: have removed as cases has been covered in DroolSscsStaffOrgRolesTest, see #1493

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        assertFalse(roleAssignments.isEmpty());
+        assertEquals(expectedRoles.split(",").length, roleAssignments.size());
+        assertThat(roleAssignments.stream().map(RoleAssignment::getRoleName).collect(Collectors.toList()),
+                containsInAnyOrder(expectedRoles.split(",")));
+        roleAssignments.forEach(r -> {
+            assertEquals("LEGAL_OPERATIONS", r.getRoleCategory().toString());
+            assertEquals("ORGANISATION", r.getRoleType().toString());
+        });
+
+        List<String> roleNamesWithRegionAttribute = List.of("tribunal-caseworker");
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java:243]: updated to includes other roles with regions in #1493

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+                "specific-access-approver-admin", "regional-centre-team-leader", "regional-centre-admin");
+
+        roleAssignments.stream().filter(c -> c.getGrantType().equals(GrantType.STANDARD)).toList()
+                .forEach(r -> {
+                    assertEquals("SSCS", r.getAttributes().get("jurisdiction").asText());
+                    assertEquals(cap.getPrimaryLocationId(), r.getAttributes().get("primaryLocation").asText());
+                    //assert region
+                    if (roleNamesWithRegionAttribute.contains(r.getRoleName())) {
+                        assertEquals("LDN", r.getAttributes().get("region").asText());
+                    }
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java:99]: updated in #1493

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+        List<String> roleNamesWithRegionAttribute = List.of("tribunal-caseworker");
+
+        roleAssignments.stream().filter(c -> c.getGrantType().equals(GrantType.STANDARD)).toList()
+                .forEach(r -> {
+                    assertEquals("SSCS", r.getAttributes().get("jurisdiction").asText());
+                    assertEquals(cap.getPrimaryLocationId(), r.getAttributes().get("primaryLocation").asText());
+                    //assert region
+                    if (roleNamesWithRegionAttribute.contains(r.getRoleName())) {
+                        assertEquals("LDN", r.getAttributes().get("region").asText());
+                    }
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java:252]: updated in #1493

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java:259]: Comment transfered to #1496

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolAdminOrgRoleMappingTest.java:309]: Comment transfered to #1496

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java:99]: Comment transfered to #1496

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java:243]: Comment transfered to #1496

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolSscsStaffOrgRolesTest.java:252]: Comment transfered to #1496

**Pull request merged**